docs/core/integration-quality-scale/rules/iqs009.md

```diff
@@ -0,0 +1,88 @@
+
+## Reasoning
+
+When a device or service is not reachable, the entities will usually go to unavailable.
+To allow the user to find out why this is happening, we should log a warning when this happens.
+Be sure to log a warning only once in total to avoid spamming the logs.
+
+When the device or service is reachable again, we should log that as well.
+This can prove useful for using the logs to find out when the device or service was unavailable and when it was back online.
+
+## Example implementation
+
+Since there are many different ways this can be implemented, we will only provide the example for entities updating via `async_update` and for integrations using the coordinator.
+
+### Example for entities updating via `async_update`
+
+In this example, we have a sensor that updates its value via `async_update`.
+We will log a warning when the sensor is unavailable and log an info message when the sensor is back online.
+Do note that we keep track of the warning being logged to avoid spamming the logs.
+
+`sensor.py`
+```python
+class MySensor(SensorEntity):
+
+    _warning_logged: bool = False
+
+    async def async_update(self) -> None:
+        try:
+            data = await self.client.get_data()
+        except MyException as ex:
+            self._attr_available = False
+            if not self._warning_logged:
+                _LOGGER.warning("The sensor is unavailable: %s", ex)
+                self._warning_logged = True
+        else:
+            self._attr_available = True
+            self._attr_native_value = data.value
+            if self._unavailable_warning_logged:
+                _LOGGER.info("The sensor is back online")
+                self._unavailable_warning_logged = False
+```
+
+### Example for integrations using the coordinator
+
+In this example, we have an integration that uses a coordinator to fetch data.
+The coordinator has the logic for logging once built in.
+The only thing that you need to do in the coordinator is to raise `UpdateFailed` when the device or service is unavailable.
+
+`coordinator.py`
+```python
+class MyCoordinator(DataUpdateCoordinator[MyData]):
+    """Class to manage fetching data."""
+
+    def __init__(self, hass: HomeAssistant, client: MyClient) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            logger=LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(minutes=1),
+        )
+        self.client = client
+    
+    async def _async_update_data(self) -> MyData:
+        try:
+            return await self.client.get_data()
+        except MyException as ex:
+            raise UpdateFailed(f"The device is unavailable: {ex}")
+```
+
+## Additional resources
+
+For more information about managing integration state, see the [documentation](../../../integration_fetching_data)
+
+## Exceptions
+
+There are no exceptions to this rule.
```

Comment:
I am wondering if there are exceptions for this rule. I can think of push based integrations that have trouble keeping track of actually being connected to the device

Reply:
I don't think that matter much, if they can't track it, this rule doesn't apply as there is nothing to log...